lte/gateway/c/core/oai/tasks/amf/amf_app_state_converter.cpp

```diff
@@ -208,4 +208,301 @@ void AmfNasStateConverter::proto_to_ue_m5gmm_context(
+  qos_flow_parameters_proto->set_preemption_capability(
+      state_qos_flow_parameters.alloc_reten_priority.pre_emption_cap);
+}
+
+void AmfNasStateConverter::proto_to_qos_flow_level_parameters(
+    const magma::lte::oai::QosFlowParameters& qos_flow_parameters_proto,
+    qos_flow_level_qos_parameters* state_qos_flow_parameters) {
+  state_qos_flow_parameters->qos_characteristic.non_dynamic_5QI_desc.fiveQI =
+      qos_flow_parameters_proto.fiveqi();
+  state_qos_flow_parameters->alloc_reten_priority.priority_level =
+      qos_flow_parameters_proto.priority_level();
+  state_qos_flow_parameters->alloc_reten_priority.pre_emption_vul =
+      static_cast<pre_emption_vulnerability>(
+          qos_flow_parameters_proto.preemption_vulnerability());
+  state_qos_flow_parameters->alloc_reten_priority.pre_emption_cap =
+      static_cast<pre_emption_capability>(
+          qos_flow_parameters_proto.preemption_capability());
+}
+
+void AmfNasStateConverter::qos_flow_setup_request_item_to_proto(
+    const qos_flow_setup_request_item& state_qos_flow_request_item,
+    magma::lte::oai::M5GQosFlowItem* qos_flow_item_proto) {
+  qos_flow_item_proto->set_qfi(state_qos_flow_request_item.qos_flow_identifier);
+  AmfNasStateConverter::qos_flow_level_parameters_to_proto(
+      state_qos_flow_request_item.qos_flow_level_qos_param,
+      qos_flow_item_proto->mutable_qos_flow_param());
+}
+
+void AmfNasStateConverter::proto_to_qos_flow_setup_request_item(
+    const magma::lte::oai::M5GQosFlowItem& qos_flow_item_proto,
+    qos_flow_setup_request_item* state_qos_flow_request_item) {
+  state_qos_flow_request_item->qos_flow_identifier = qos_flow_item_proto.qfi();
+  AmfNasStateConverter::proto_to_qos_flow_level_parameters(
+      qos_flow_item_proto.qos_flow_param(),
+      &state_qos_flow_request_item->qos_flow_level_qos_param);
+}
+// smf_context to proto and proto to smf_context
+void AmfNasStateConverter::smf_context_to_proto(
+    const smf_context_t* state_smf_context,
+    magma::lte::oai::SmfContext* smf_context_proto) {
+  smf_context_proto->set_sm_session_state(state_smf_context->pdu_session_state);
+  smf_context_proto->set_pdu_session_version(
+      state_smf_context->pdu_session_version);
+  smf_context_proto->set_active_pdu_sessions(state_smf_context->n_active_pdus);
+  smf_context_proto->set_is_emergency(state_smf_context->is_emergency);
+  AmfNasStateConverter::session_ambr_to_proto(
+      state_smf_context->selected_ambr,
+      smf_context_proto->mutable_selected_ambr());
+
+  smf_context_proto->set_gnb_gtp_teid(
+      state_smf_context->gtp_tunnel_id.gnb_gtp_teid);
+
+  char gnb_gtp_teid_ip_addr_str[16] = {0};
+  inet_ntop(
+      AF_INET, state_smf_context->gtp_tunnel_id.gnb_gtp_teid_ip_addr,
+      gnb_gtp_teid_ip_addr_str, INET_ADDRSTRLEN);
+  smf_context_proto->set_gnb_gtp_teid_ip_addr(gnb_gtp_teid_ip_addr_str);
+
+  smf_context_proto->set_upf_gtp_teid(
+      *(uint32_t*) &state_smf_context->gtp_tunnel_id.upf_gtp_teid);
```

Comment:
Why do we need the pointer creation and dereferencing?
```suggestion
      state_smf_context->gtp_tunnel_id.upf_gtp_teid);
```

Reply:
Addressed this review comment.